transport_versioned_docs/version-2.0.0/migration.md

```diff
@@ -1,9 +1,46 @@
+
+  * There is no concept of `NetworkInterfaceEndPoint` anymore. It has been completely replaced with the more general `NetworkEndpoint`. Consequently, there is no need to implement conversion logic between the two anymore (so `CreateInterfaceEndPoint` and `GetGenericEndPoint` were removed from `INetworkInterface`).
+  * There is no need to provide a `NetworkSendInterface` through `CreateSendInterface` anymore. Send operations are now handled entirely by `ScheduleSend`, which gets passed a `PacketsQueue` containing the packets to be sent.
+  * The `ScheduleReceive` method doesn't use `NetworkPacketReceiver` (which is now obsolete) to propagate received packets to the rest of UTP. Instead, implementations of `ScheduleReceive` are expected to fill the `PacketsQueue` that is now passed in with the received packets.
+  * Implementations of `INetworkInterface` are now expected to be fully unmanaged. However, a managed implementation can be wrapped into an unmanaged one with the new `WrapToUnmanaged` extension method.
+  * Creating a `NetworkDriver` with a custom network interface must now be done using the static `NetworkDriver.Create` method (e.g. `NetworkDriver.Create(new MyCustomInterface())`). Directly constructing a `NetworkDriver` with `new` is deprecated.
+
+The reason for these breaking changes is to simplify the interface and make it more flexible. For more details on how to create custom network interfaces, please refer to [this section of the documentation](custom-network-interface.md).
+
+## Pipeline stages
+
+There are no changes to how custom pipeline stages themselves are written. However, the mechanisms used to register them and get the identifier of a pipeline stage are different:
+
+  * Instead of registering a custom pipeline stage with `NetworkPipelineStageCollection.RegisterPipelineStage`, use `NetworkDriver.RegisterPipelineStage`. Note that this must be performed on every instance of `NetworkDriver` that will make use of the custom pipeline stage.
+  * Instead of retrieving the ID of a pipeline stage with `NetworkPipelineStageCollection.GetStageId`, use the static `NetworkPipelineStageId.Get` method.
+
+The reason for these breaking changes is to remove Burst-incompatible APIs, allowing more of UTP to be used from Burst-compiled code. For more details on how to create custom pipeline stage, please refer to [this section of the documentation](custom-pipeline.md). For more details on pipelines in general, see [this section instead](pipelines-usage.md).
+
+## Other breaking changes
+
+  * After calling `NetworkDriver.Disconnect`, notifying the remote peer of the disconnection now requires completing a `NetworkDriver.ScheduleUpdate` job. In 1.X, `NetworkDriver.ScheduleFlushSend` used to be sufficient for this purpose but it's not the case anymore.
```

Comment:
All the document is factual without explanation, I think this is the correct approach. 
However, for this, we are adding an extra step for the user. The natural reaction will be to wonder why. 
Would an explanation make sense here ?

Reply:
@johann-r-unity - Forgive me for lacking context here; I'm not sure I understand entirely. Are you saying we should add an explanation around why we introduced these additional breaking changes (similar to the explanation at the end of the previous paragraphs)?

For example:
> These breaking changes simplify and increase the flexibility of the interface. Please refer to this documentation section for more details on creating custom network interfaces.

Reply:
Hi Larah, 
My initial comment is outdated. Simon did the required modification already.